core/vtk/ttkMetricDistortion/ttkMetricDistortion.cpp

```diff
@@ -0,0 +1,250 @@
+#include <ttkMetricDistortion.h>
+
+#include <vtkInformation.h>
+
+#include <vtkCellData.h>
+#include <vtkDataArray.h>
+#include <vtkDataSet.h>
+#include <vtkDoubleArray.h>
+#include <vtkObjectFactory.h>
+#include <vtkPointData.h>
+#include <vtkPolyData.h>
+#include <vtkSmartPointer.h>
+#include <vtkTable.h>
+
+#include <ttkMacros.h>
+#include <ttkUtils.h>
+
+// A VTK macro that enables the instantiation of this class via ::New()
+// You do not have to modify this
+vtkStandardNewMacro(ttkMetricDistortion);
+
+/**
+ * Implement the filter constructor and destructor in the cpp file.
+ *
+ * The constructor has to specify the number of input and output ports
+ * with the functions SetNumberOfInputPorts and SetNumberOfOutputPorts,
+ * respectively. It should also set default values for all filter
+ * parameters.
+ *
+ * The destructor is usually empty unless you want to manage memory
+ * explicitly, by for example allocating memory on the heap that needs
+ * to be freed when the filter is destroyed.
+ */
+ttkMetricDistortion::ttkMetricDistortion() {
+  this->SetNumberOfInputPorts(2);
+  this->SetNumberOfOutputPorts(1);
+}
+
+ttkMetricDistortion::~ttkMetricDistortion() {
+}
+
+/**
+ * Specify the required input data type of each input port
+ *
+ * This method specifies the required input object data types of the
+ * filter by adding the vtkAlgorithm::INPUT_REQUIRED_DATA_TYPE() key to
+ * the port information.
+ */
+int ttkMetricDistortion::FillInputPortInformation(int port,
+                                                  vtkInformation *info) {
+  if(port == 0) {
+    info->Set(vtkAlgorithm::INPUT_REQUIRED_DATA_TYPE(), "vtkPolyData");
```

Comment:
Side note for later: we may want to change this to `vtkPointSets` if we want to generalize to `vtkUnstructuredGrid` (to generalize the same kind of features to tetrahedral meshes).

Reply:
Done in the next commit.